From d90faa7bd7ce3b415032dd193f7bd4acfe1617f5 Mon Sep 17 00:00:00 2001
From: Oleksandr Natalenko <oleksandr@natalenko.name>
Date: Tue, 1 Nov 2022 18:39:59 +0100
Subject: [PATCH 2/2] v4l2loopback: un-confine v4l2loopback_cleanup_module()

@TheArcaneBrony reported this happening when building the `v4l2loopback`
as a built-in:

```
drivers/media/v4l2-core/v4l2loopback.c:2969:13: error: use of undeclared identifier 'v4l2loopback_cleanup_module'; did you mean 'v4l2loopback_init_module'?
```

The reason for this is that 550c240585 introduced extra `ifdef MODULE`
around `v4l2loopback_cleanup_module()`, but subsequently 1fcd767eb3
opted into using `module_init()`/`module_exit()` instead, which made
the `ifdef` excessive.

Hence, remove extra `ifdef MODULE`.

Fixes: 1fcd767eb3 ("use module_init()/module_exit() rather than '#ifdef MODULE'")
Reported-by: The Arcane Brony <myrainbowdash949@gmail.com>
Tested-by: The Arcane Brony <myrainbowdash949@gmail.com>
Signed-off-by: Oleksandr Natalenko <oleksandr@natalenko.name>
---
 drivers/media/v4l2-core/v4l2loopback.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/media/v4l2-core/v4l2loopback.c b/drivers/media/v4l2-core/v4l2loopback.c
index d5dfe2c28..ed6cfea78 100644
--- a/drivers/media/v4l2-core/v4l2loopback.c
+++ b/drivers/media/v4l2-core/v4l2loopback.c
@@ -2951,7 +2951,6 @@ static int __init v4l2loopback_init_module(void)
 	return err;
 }
 
-#ifdef MODULE
 static void v4l2loopback_cleanup_module(void)
 {
 	MARK();
@@ -2961,7 +2960,6 @@ static void v4l2loopback_cleanup_module(void)
 	misc_deregister(&v4l2loopback_misc);
 	dprintk("module removed\n");
 }
-#endif
 
 MODULE_ALIAS_MISCDEV(MISC_DYNAMIC_MINOR);
 
-- 
2.38.1.143.g1fc3c0ad40

